prog/app.go

```diff
@@ -40,14 +42,81 @@ func appMain() {
 		weaveHostname  = flag.String("weave.hostname", app.DefaultHostname, "Hostname to advertise in WeaveDNS")
 		containerName  = flag.String("container.name", app.DefaultContainerName, "Name of this container (to lookup container ID)")
 		dockerEndpoint = flag.String("docker", app.DefaultDockerEndpoint, "Location of docker endpoint (to lookup container ID)")
+
+		collectorType     = flag.String("collector", "local", "Collector to use (local of dynamodb)")
+		controlRouterType = flag.String("control.router", "local", "Control router to use (local or sqs)")
+		pipeRouterType    = flag.String("pipe.router", "local", "Pipe router to use (local)")
+		userIDHeader      = flag.String("userid.header", "", "HTTP header to use as userid")
+
+		awsDynamoDB     = flag.String("aws.dynamodb", "", "URL of DynamoDB instance")
+		awsCreateTables = flag.Bool("aws.create.tables", false, "Create the tables in DynamoDB")
+		awsSQS          = flag.String("aws.sqs", "", "URL of SQS instance")
+		awsRegion       = flag.String("aws.region", "", "AWS Region")
+		awsID           = flag.String("aws.id", "", "AWS Account ID")
+		awsSecret       = flag.String("aws.secret", "", "AWS Account Secret")
+		awsToken        = flag.String("aws.token", "", "AWS Account Token")
+
+		consulPrefix = flag.String("consul.prefix", "", "Prefix for keys in consul")
+		consulAddr   = flag.String("consul.addr", "", "Address of consul instance")
+		consulInf    = flag.String("consul.inf", "", "The interface who's address I should advertise myself under in consul")
```

Comment:
> adding all these flags to the open source product seems mediocre.

This was the least-worse option.

> What about consolidating them into a URL?

Whilst this would be slightly nicer, it doesn't solve the real problem: the options will still exist.  I suggest we just live with it, potentially marking them as "experimental" and hiding them from `help` by default (maybe unless someone does a `-hhh` or something, like `-vvv`)

Also, I don't want to write no skanky URL parsing code...